src/UserManager.ts

```diff
@@ -424,7 +424,11 @@ export class UserManager {
             const signinRequest = await this._client.createSigninRequest(args);
             logger.debug("got signin request");
 
-            const url = await signinRequest.getUrl();
+            const url = signinRequest.url;
+
+            if (!url) {
```

Comment:
it guess its more safe to move the throw in the `url` getter and return always a `string`. This way we do not need any change in this file...

Reply:
as far as the type system's concerned, you could forget to call `setUrl`, though. which is why `this.url` can be `undefined`

we happen to know we call it, though. so maybe we could just default `this._url` to `''` for sake of it always being a string (since we'll have always called `setUrl` to replace that anyway)

Reply:
ah nevermind, i get you! i misread 👍